### ISSUE TITLE
TargetFramework net6.0-windows10.0.19041.0 does not work

### ISSUE THREAD

**Issue by amertlich** [Azure/azure-mobile-apps#576]
## Describe the bug

Using both WPF or Test projects targeting net6.0-windows10.0.19041.0 does not appear to work.  It seems to assume a WinRT platform path based on the stack trace.

```
System.InvalidOperationException : Exception of type 'System.InvalidOperationException' was thrown.
Data:
  Description: <null>
  RestrictedDescription: <null>
  RestrictedErrorReference: <null>
  RestrictedCapabilitySid: <null>
  __RestrictedErrorObjectReference: <null>
  __HasRestrictedLanguageErrorObject: False
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Windows.Storage.IApplicationDataStaticsMethods.get_Current(IObjectReference _obj)
   at Windows.Storage.ApplicationData.get_Current()
   at Microsoft.Datasync.Client.Platforms.ApplicationStorage..ctor(String containerName)
   at Microsoft.Datasync.Client.Platforms.CurrentPlatform.<>c.<.ctor>b__6_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Datasync.Client.Platforms.CurrentPlatform.get_ApplicationStorage()
   at Microsoft.Datasync.Client.Utils.Platform.GetApplicationInstallationId()
   at Microsoft.Datasync.Client.Utils.Platform.<>c.<.cctor>b__12_1()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Datasync.Client.Utils.Platform.get_InstallationId()
   at Microsoft.Datasync.Client.Http.ServiceHttpClient..ctor(Uri endpoint, AuthenticationProvider authenticationProvider, DatasyncClientOptions clientOptions)
   at Microsoft.Datasync.Client.DatasyncClient..ctor(Uri endpoint, AuthenticationProvider authenticationProvider, DatasyncClientOptions clientOptions)
   at Microsoft.Datasync.Client.DatasyncClient..ctor(Uri endpoint, DatasyncClientOptions clientOptions)
```

## To Reproduce

Change the .csproj `TargetFramework` to use net6.0-windows10.0.19041.0.  Construct a new instance of the `Microsoft.Datasync.Client.DatasyncClient` and an exception is thrown.

## Expected behavior

Should be able to use the DatasyncClient in desktop apps leveraging Windows Runtime APIs

## What platforms?

* Server (ASP.NET Framework, ASP.NET Core, NodeJS):
n/a

* Client (.NET, JavaScript, iOS, Android):
WPF
net6.0-windows10.0.19041.0
Haven't tried other targets using the WinRT APIs in Desktop apps, but my guess is other targets may be broken too.

**Comment by adrianhall** [comment]
This is going to be a bit of a bear.  See https://github.com/microsoft/WindowsAppSDK/issues/1721

Basically, you need to package to be able to support ApplicationStorage.

I'm going to have to re-think the whole ApplicationStorage thing and how it is manifest in the product, which will take some time.

**Comment by adrianhall** [comment]
In the interim, use ClientOptions.InstallationId to set a specific installation ID:

```csharp
var options = new DatasyncClientOptions { InstallationId = "test" };
var client = new DatasyncClient(serviceUri, options);
```

This can be set rather easily in the TodoApp.Data project - add it to the default options handler at line 91 in the RemoteTodoService.cs file.

**Comment by adrianhall** [comment]
The fix is in.  Basically, if you have an unpackaged app, then the ApplicationDataStore is not available and you SHOULD set the InstallationId accordingly.  If you don't, then a unique ID is set and used instead.  This is "as good as it gets" for the setup.

Will be in next release.

**Comment by amertlich** [comment]
Seems fine enough.  Although I'm not sure what you're using the InstallationId for?  Is it critical? Is it just used in the HTTP request headers for client request tracking server side?

**Comment by adrianhall** [comment]
Yeah - it's for client request tracking server side.

**Comment by adrianhall** [comment]
"Fixed" in #582
